IPython/utils/path.py

```diff
@@ -443,81 +367,19 @@ def target_update(target,deps,cmd):
 
-# If the config is unmodified from the default, we'll just delete it.
-# These are consistent for 0.10.x, thankfully. We're not going to worry about
-# older versions.
-old_config_md5 = {'ipy_user_conf.py': 'fc108bedff4b9a00f91fa0a5999140d3',
-                  'ipythonrc': '12a68954f3403eea2eec09dc8fe5a9b5'}
-
-def check_for_old_config(ipython_dir=None):
-    """Check for old config files, and present a warning if they exist.
-
-    A link to the docs of the new config is included in the message.
-
-    This should mitigate confusion with the transition to the new
-    config system in 0.11.
-    """
-    if ipython_dir is None:
-        ipython_dir = get_ipython_dir()
-
-    old_configs = ['ipy_user_conf.py', 'ipythonrc', 'ipython_config.py']
-    warned = False
-    for cfg in old_configs:
-        f = os.path.join(ipython_dir, cfg)
-        if os.path.exists(f):
-            if filehash(f) == old_config_md5.get(cfg, ''):
-                os.unlink(f)
-            else:
-                warn("Found old IPython config file {!r} (modified by user)".format(f))
-                warned = True
-
-    if warned:
-        warn("""
-  The IPython configuration system has changed as of 0.11, and these files will
-  be ignored. See http://ipython.github.com/ipython-doc/dev/config for details
-  of the new config system.
-  To start configuring IPython, do `ipython profile create`, and edit
-  `ipython_config.py` in <ipython_dir>/profile_default.
-  If you need to leave the old config files in place for an older version of
-  IPython and want to suppress this warning message, set
-  `c.InteractiveShellApp.ignore_old_config=True` in the new config.""")
-
+@undoc
 def get_security_file(filename, profile='default'):
-    """Return the absolute path of a security file given by filename and profile
-    
-    This allows users and developers to find security files without
-    knowledge of the IPython directory structure. The search path
-    will be ['.', profile.security_dir]
-    
-    Parameters
-    ----------
-    
-    filename : str
-        The file to be found. If it is passed as an absolute path, it will
-        simply be returned.
-    profile : str [default: 'default']
-        The name of the profile to search.  Leaving this unspecified
-        The file to be found. If it is passed as an absolute path, fname will
-        simply be returned.
-    
-    Returns
-    -------
-    Raises :exc:`IOError` if file not found or returns absolute path to file.
-    """
-    # import here, because profiledir also imports from utils.path
-    from IPython.core.profiledir import ProfileDir
-    try:
-        pd = ProfileDir.find_profile_dir_by_name(get_ipython_dir(), profile)
-    except Exception:
-        # will raise ProfileDirError if no such profile
-        raise IOError("Profile %r not found")
-    return filefind(filename, ['.', pd.security_dir])
+    warn("get_security_file has moved to the IPython.paths module")
```

Comment:
get_security_file is going to leave IPython entirely, since it's a Jupyter function.